fem/pfespace.cpp

```diff
@@ -2553,24 +2535,31 @@ ParFiniteElementSpace::ParallelDerefinementMatrix(int old_ndofs,
    int nrk = HYPRE_AssumedPartitionCheck() ? 2 : NRanks;
 
    MFEM_VERIFY(Nonconforming(), "Not implemented for conforming meshes.");
-   MFEM_VERIFY(pmesh->GetNumGeometries(pmesh->Dimension()) == 1,
-               "Not implemented for mixed meshes.");
-
    MFEM_VERIFY(old_dof_offsets[nrk], "Missing previous (finer) space.");
+
+#if 0 // check no longer seems to work with NC tet refinement
    MFEM_VERIFY(dof_offsets[nrk] <= old_dof_offsets[nrk],
                "Previous space is not finer.");
+#endif
 
    // Note to the reader: please make sure you first read
    // FiniteElementSpace::RefinementMatrix, then
    // FiniteElementSpace::DerefinementMatrix, and only then this function.
    // You have been warned! :-)
 
+   Mesh::GeometryList elem_geoms(*mesh);
+
    Array<int> dofs, old_dofs, old_vdofs;
    Vector row;
 
    ParNCMesh* pncmesh = pmesh->pncmesh;
-   Geometry::Type geom = pncmesh->GetElementGeometry(0); // TODO mixed meshes
-   int ldof = fec->FiniteElementForGeometry(geom)->GetDof();
+
+   int ldof[Geometry::NumGeom] = { 0 };
```

Comment:
Hmm, is this supposed to initialize all entries of the array? I'm not sure this is what happens with this initializer.

Reply:
This is standard in C99 and should work in C++ too (where it actually could read `... = {};` but the C style seems more explicit). I can change this if you want.

Reply:
I put back the branch `ref_algo == 1 && meshgen == 1`. The branch is now at the same level (in the method) as the other cases, so it doesn't get lost in the future (that's why the refinement list code is duplicated).

Reply:
Made the initialization explicit...